src/compiler/checker.ts

```diff
@@ -6882,7 +7093,7 @@ module ts {
             // reorderCandidates fills up the candidates array directly
             reorderCandidates(signatures, candidates);
             if (!candidates.length) {
-                error(node, Diagnostics.Supplied_parameters_do_not_match_any_signature_of_call_target);
+                reportError(Diagnostics.Supplied_parameters_do_not_match_any_signature_of_call_target);

```

Comment:
`reportError` wraps the chained diagnostic for the decorator, see line 7229 in the modified source below. It is used in a few places, so I wrapped the logic in a function to remove redundancy.